### PR TITLE
fix #11 - wrap ajax in Promises

### LINKED ISSUE #11: Add-on only reports for a single stylesheet in the page
Steps to reproduce:
1. Add a site to the dev domains (e.g., developer.allizom.org)
2. Go to a page on the site with 2 stylesheets (e.g., https://developer.allizom.org/en-US/ has mdn.css and home.css)

Expected results:
Should receive an alert for each stylesheet's compatibility issues

Actual results:
Only receive a single alert

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

**Comment by openjck** [comment]
Just a question about workflow. Should the _.xpi_ be re-built with this commit or is the idea to continually change the source and re-build the package as part of a special release/deploy process? Either way is fine. Just want to be sure to know what to look for.

**Comment by openjck** [comment]
Didn't mean to close. Reopening...

File: data/doiuse-script.js [hunk 1]
```diff
@@ -1,42 +1,51 @@
...
+      return core.ajax('GET', url);
+    },
+    'post': function(data) {
+      return core.ajax('POST', url, data);
+    }
+  };
+}
+
+function postToDoiuse (css) {
+  var styleSheetContent = css;
```
Comment (openjck) [inline data/doiuse-script.js]: Not a big deal, but why the rename?

**Comment by openjck** [comment]
Also, this is very cool. I'm a fan.

**Comment by groovecoder** [comment]
Updated.

Good point on the `.xpi` file. I updated the file in this PR since it's already in the repo. In the future, yes - we would compile the `.xpi` file as part of a build/deploy process, and (hopefully) automatically upload it to addons.mozilla.org.

**Comment by openjck** [comment]
Looks good, works as expected r+

**Pull request merged**